core/admin/mailu/sso/views/base.py

```diff
@@ -50,11 +74,22 @@ def login():
             utils.limiter.rate_limit_user(username, client_ip, device_cookie, device_cookie_username) if models.User.get(username) else utils.limiter.rate_limit_ip(client_ip)
             flask.current_app.logger.warn(f'Login failed for {username} from {client_ip}.')
             flask.flash('Wrong e-mail or password', 'error')
-    return flask.render_template('login.html', form=form, fields=fields)
+    return flask.render_template('login.html', form=form, fields=fields, openId=app.config['OIDC_ENABLED'], openIdEndpoint=utils.oic_client.get_redirect_url())
 
 @sso.route('/logout', methods=['GET'])
 @access.authenticated
 def logout():
+    if utils.oic_client.is_enabled():
+        if 'openid_token' not in flask.session:
+            return logout_legacy()
+        if 'state' in flask.request.args and 'state' in flask.session:
```

Comment:
This is changing the existing logic: we currently do not protect against "forced logouts" and have no CSRF protection here by choice.

Why do you want to change this?